### PR TITLE
client: Remove unused SetLeaves method

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,7 +15,8 @@ to the experimental status of the Map.
 The map client has been updated so that GetAndVerifyMapLeaves and
 GetAndVerifyMapLeavesByRevision return the MapRoot for the revision at which the
 leaves were fetched. Without this callers of GetAndVerifyMapLeaves in particular
-were unable to reason about which map revision they were seeing.
+were unable to reason about which map revision they were seeing. The unused
```
Comment (mhutchinson) [inline CHANGELOG.md]: Drop the unused. It was used in the last version of Trillian we published.

**Comment by pavelkalinnikov** [comment]
> LGTM. My one concern is whether we should create a map_write_client.go before we remove this?

I don't think it should be strictly before. I suggest to add it when it's actually needed, as right now things seem to be working without it.

**Comment by Martin2112** [comment]
I think we can probably do this before the map_write_client.

**Comment by Martin2112** [comment]
Travis doesn't seem happy.

**Comment by pavelkalinnikov** [comment]
Oups, I mixed it with another branch.

**Pull request merged**